### PR TITLE
Everyones A Crytic: Start Crytic integration

### PR DESCRIPTION
**Pull request by Shadowfiend**
This PR adds full support for the Crytic GitHub app. It configures it
to run Slither on our `solidity` directory, and configures Slither to ignore
test stubs and run all detectors except `assembly` and `timestamp`
(which we choose to leave to ethlint), `solc-version`(which has outdated
version recommendations at the moment), and `naming-convention` (which
has valid recommendations that we are choosing to hold off on for v1).

This PR also fixes a variety of mostly cosmetic issues that Crytic caught
throughout the codebase. The most notable change here is a rework of
operation ordering in `TBTCSystem.calculateBondRequirementWei` that
offers greater precision for the final bond requirement. The ultimate impact
should be fairly negligible, but it's a win regardless.

### CODE REVIEW

**Comment by Shadowfiend** [comment]
I'm disabling `naming-convention` here and marking this ready for merge. `naming-convention` is actually something I think we should look at, but it will have to wait until 2.0 since we're so deep into our current naming conventions.

File: solidity/contracts/system/TBTCSystem.sol [hunk 1]
```diff
@@ -625,48 +626,46 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
...
     function calculateBondRequirementWei(
-        uint256 _lotSizeSatoshis
+        uint256 _requestedLotSizeSatoshis
     ) internal view returns (uint256) {
-        uint256 bondRequirementSatoshis = _lotSizeSatoshis.mul(
-            initialCollateralizedPercent
-        ).div(100);
-        return _fetchBitcoinPrice().mul(bondRequirementSatoshis);
+        uint256 lotSizeInWei = _fetchBitcoinPrice().mul(_requestedLotSizeSatoshis);
+        return lotSizeInWei.mul(initialCollateralizedPercent).div(100);
```
Comment (pdyraga) [inline solidity/contracts/system/TBTCSystem.sol:668]:
```
btcPrice * (lotSizeSatoshis * initialCollateralizedPercent / 100)
== 
btcPrice * lotSizeSatoshis * initialCollateralizedPercent / 100
```

👍 

b228138ca189b63989d295492e8a8b16

**Review by pdyraga — APPROVED** [review]
👍

File: solidity/contracts/deposit/DepositLiquidation.sol [hunk 2]
```diff
@@ -22,25 +22,25 @@ library DepositLiquidation {
...
         uint8 _v,
         bytes32 _r,
         bytes32 _s,
         bytes32 _signedDigest,
         bytes memory _preimage
-    ) public returns (bool _isFraud) {
+    ) public {
         IBondedECDSAKeep _keep = IBondedECDSAKeep(_d.keepAddress);
-        return _keep.submitSignatureFraud(_v, _r, _s, _signedDigest, _preimage);
+        _keep.submitSignatureFraud(_v, _r, _s, _signedDigest, _preimage);
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply solidity/contracts/deposit/DepositLiquidation.sol:43]: Nope, I missed the fact it's possible slashing failed and we don't revert in this case. This boolean returned from the function could be useful.

File: .gitignore [hunk 3]
```diff
@@ -14,6 +14,7 @@ implementation/build/
 blocks/
 __pycache__/
 node_modules/
+artifacts/
```
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply .gitignore:17]: … Probably at some point yeah. I haven't run into issues here, but I may just have been distracted.

File: solidity/contracts/deposit/DepositLiquidation.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply solidity/contracts/deposit/DepositLiquidation.sol:43]: Are you sure? https://github.com/keep-network/keep-ecdsa/blob/master/solidity/contracts/AbstractBondedECDSAKeep.sol#L382 returns `isFraud`, but https://github.com/keep-network/keep-ecdsa/blob/master/solidity/contracts/AbstractBondedECDSAKeep.sol#L374 reverts if `isFraud` is `true`. None of the intervening code updates `isFraud`. `slashForSignatureFraud` does not return anything, it simply emits `SlashingFailed` if something went wrong.

**Pull request merged**